### PR TITLE
Quml odk converter

### PR DESCRIPTION
**Pull request by choxx**

### CODE REVIEW

File: src/quml-to-odk/quml-to-odk.service.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+      // let's sort the available questions in random order
+      const randomQuestionsList = response.result.Question.sort(
+        () => Math.random() - 0.5,
+      );
+      questionIdentifiers = randomQuestionsList
+        .slice(0, filters.randomQuestionsCount)
+        .map((obj) => {
+          return obj.identifier;
+        });
+      console.log(
```
Comment (ChakshuGautam) [inline src/quml-to-odk/quml-to-odk.service.ts:82]: Remove console.log. Use logger everywhere.

File: test/app.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from './../src/app.module';
+
+describe('AppController (e2e)', () => {
```
Comment (ChakshuGautam) [inline test/app.e2e-spec.ts:6]: Add a test with sample questions and see if the XML is getting created perfectly.

File: src/quml-to-odk/quml-to-odk.service.ts [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+   *
+   * @return string file (absolute path of the generated XSLX form)
+   */
+  private createXslx(
+    surveySheetArray: Array<any>,
+    choicesSheetArray: Array<any>,
+    mediaSheetArray: Array<any>,
+    settingsSheetArray: Array<any>,
+  ): string {
+    const filename = +new Date() + '.xlsx';
```
Comment (ChakshuGautam) [inline src/quml-to-odk/quml-to-odk.service.ts:381]: create a UUID based filename.

File: src/quml-to-odk/quml-to-odk.service.ts [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+  }
+
+  private async fetchQuestions(filters: GenerateFormDto): Promise<any> {
+    const requestBody = {
+      request: {
+        filters: {
+          se_boards: filters.boards,
+          gradeLevel: filters.grades,
+          subject: filters.subjects,
+          qType: 'MCQ', // we'll always fetch MCQ questions
```
Comment (ChakshuGautam) [inline src/quml-to-odk/quml-to-odk.service.ts:52]: Make this a config. We will expand to questions other than MCQ as well.

File: src/quml-to-odk/quml-to-odk.service.ts [hunk 5]
```diff
@@ -0,0 +1,404 @@
+import { Injectable } from '@nestjs/common';
+import { GenerateFormDto } from './dto/generate-form.dto';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
+import { exec } from 'child_process';
+import * as XLSX from 'xlsx';
```
[the remark this answers is not in the document]
Comment (choxx) [reply src/quml-to-odk/quml-to-odk.service.ts:6]: Using sheetjs only: referring to their documentation as available here: https://docs.sheetjs.com/docs/installation/nodejs#usage

File: src/quml-to-odk/quml-to-odk.service.ts (hunk 1 above)
Comment (choxx) [reply src/quml-to-odk/quml-to-odk.service.ts]: It's done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/37409db0ef09a023ab3b16b2ba6a68dc308286f0

File: src/quml-to-odk/quml-to-odk.service.ts (hunk 3 above)
Comment (choxx) [reply src/quml-to-odk/quml-to-odk.service.ts]: It's done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/41c603d1e89e549c052e409f47a60e9ffa6b6f4e

File: test/app.e2e-spec.ts (hunk 2 above)
Comment (choxx) [reply test/app.e2e-spec.ts:7]: I added the test case [here](https://github.com/Samagra-Development/quml2xform/pull/1/commits/b2c92266da98450540829216b1439a53c5c9f282) to check if the newly generated form files exist for real by upon hitting the newly added route.

File: src/quml-to-odk/quml-to-odk.service.ts [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+
+@Injectable()
+export class QumlToOdkService {
+  // config file for the variables being used across the service
+  private config = {
+    questionBankUrl: process.env.QUML_ODK_QUESTION_BANK_URL,
```
[the remark this answers is not in the document]
Comment (choxx) [reply src/quml-to-odk/quml-to-odk.service.ts]: It's done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/12083cc01bf6508ecb0e54970267d787d117ac7c

File: src/quml-to-odk/quml-to-odk.service.ts (hunk 4 above)
Comment (choxx) [reply src/quml-to-odk/quml-to-odk.service.ts]: Done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/cd3caad3799af0c85e39301f07f8d50e75d9755c

**Pull request merged**